Client/IRequest.cs

```diff
@@ -84,6 +84,10 @@ public interface IRequest
         /// <param name="enabled">Whether to raise HTTP errors as exceptions.</param>
         IRequest WithHttpErrorAsException(bool enabled);
 
+        /// <summary>Set whether aguments with null value should be ignored for this request.</summary>
+        /// <param name="ignoreNullArguments">Whether to ignore arguments with null value.</param>
+        IRequest WithIgnoreNullArguments(bool ignoreNullArguments);
```

Comment:
I was trying to be consistent with `WithHttpErrorAsException`. I have no problem changing it, shall I also remove WithHttpErrorAsException?

Reply:
I suggest marking that one obsolete for removal in 4.0.

Reply:
ok, here's what I propose:
- Mark `IRequest.WithHttpErrorAsException` as obsolete
- Mark `IClient.SetHttpErrorAsException` as obsolete (for consistency sake)
- Add `IRequest.WithOptions`
- Add `IClient.WithOptions` (for consistency sake)

To make sure we are on same page: the will be a FluentClientOptions class as a RequestOptions class.

Reply:
Also, since the will be a `IClient.WithOptions` method, how about removing the `options` parameter from the `FluentClient` constructor (thereby restoring the constructor to what it was prior to this PR)

Reply:
That looks good, though I suggest `IClient.SetOptions` for consistency with the other client methods. (The distinction is meant to avoid confusion since they work differently: `IClient.Set*` applies for all requests, and `IRequest.With*` applies temporarily for the current request.)

Reply:
I'm fine removing the constructor argument.